### ISSUE TITLE
BigQuery `CREATE VECTOR INDEX` statement with `STORING` causes parsing error

### ISSUE THREAD

**Issue by agkphysics** [sqlfluff/sqlfluff#7102]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

The `STORING` clause of the `CREATE VECTOR INDEX` statement causes SQLFluff to throw a parsing error because it is not included in the grammar: https://github.com/sqlfluff/sqlfluff/blob/3e879b1f64fc69d649faf8bd8d302efb73ebceb8/src/sqlfluff/dialects/dialect_bigquery.py#L3124-L3131

### Expected Behaviour

Having a `STORING (...)` clause should not cause parsing issues since it is optional: https://cloud.google.com/bigquery/docs/reference/standard-sql/data-definition-language#create_vector_index_statement

### Observed Behaviour

The `STORING` clause fails with `Found unparsable section: 'CREATE VECTOR INDEX `index` ON `table` (...`

### How to reproduce

```sql
-- Works fine
CREATE VECTOR INDEX `index` ON `table` (embedding) OPTIONS (index_type = "IVF")

-- Fails
CREATE VECTOR INDEX `index` ON `table` (embedding) STORING (col1) OPTIONS (index_type = "IVF")
```

### Dialect

BigQuery

### Version

```
Python 3.12.11
sqlfluff, version 3.4.2
```

### Configuration

```toml
[tool.sqlfluff.core]
dialect = "bigquery"
max_line_length = 120
exclude_rules = ["CP02", "CP03", "RF01", "RF02", "ST06", "LT08", "ST07"]

[tool.sqlfluff.indentation]
indented_joins = false
indented_ctes = true
indent_unit = "space"
tab_space_size = 2

[tool.sqlfluff.rules.convention.quoted_literals]
preferred_quoted_literal_style = "double_quotes"

[tool.sqlfluff.rules.references.keywords]
ignore_words = "TIMESTAMP,NAME,TYPE"  # We use these as a column names

[tool.sqlfluff.rules.structure.join_condition_order]
preferred_first_table_in_join_clause = "later"
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by onestardao** [comment]
Looking at this issue, the core problem is clear: BigQuery’s `CREATE VECTOR INDEX ... STORING` syntax isn’t yet supported in the `sqlfluff` parser, which causes the parse error. This matches directly with **Problem Map No.7 – Missing grammar extensions**, where the parser simply hasn’t been updated for newer dialect features.

This isn’t an infra problem and you don’t need to rebuild the environment. The root cause is that the semantic firewall doesn’t recognize the extra keyword, so parsing fails. Many users run into this early on and think it’s complex, but once you map it to the right category it’s straightforward to handle.

If you want to test or debug quickly, you can take the exact SQL or log screenshot and ask an AI something like:

> “Use WFGY inside TXTOS to solve No.7 grammar extension gap. How should I handle this SQL?”

The AI will apply the mapped reasoning and give a targeted fix.
If you’d like, I can share the full Problem Map documentation that explains No.7 in detail with examples. Would you like me to provide the link?
